app/controllers/api/groups_controller.rb

```diff
@@ -1,6 +1,12 @@
 class API::GroupsController < API::RestfulController
   load_and_authorize_resource only: :show, find_by: :key
   load_resource only: [:upload_photo, :use_gift_subscription], find_by: :key
+  skip_before_action :authenticate_user!, only: :index
```

Comment:
Hrm, I'm surprised to see we need to do this, as we shouldn't be authing the user for group actions across the board.

What kind of error do you get if this line goes away?